### PR TITLE
Autotune Dataset.map when transforming builtin map with autograph

### PR DESCRIPTION
**Pull request by lgeiger**
This PR enables autotuning for `tf.data.Dataset.map()` calls that have been generated from built in `map()` via autograph. This has the potential to significantly increase performance in some cases.

### CODE REVIEW

**Review by mdanatg — APPROVED** [review]

File: tensorflow/python/autograph/operators/py_builtins.py [hunk 1]
```diff
@@ -427,7 +428,8 @@ def map_(fn, *iterables):
 
 
 def _tf_dataset_map(fn, *iterables):
-  return dataset_ops.DatasetV2.zip(iterables).map(fn)
+  zipped_dataset = dataset_ops.DatasetV2.zip(iterables)
+  return zipped_dataset.map(fn, num_parallel_calls=dataset_ops.AUTOTUNE)
```
Comment (jsimsa) [inline tensorflow/python/autograph/operators/py_builtins.py:432]: this is generally not correct because it does not preserve semantics of stateful map functions

please revert this change

Comment (mdanatg) [reply tensorflow/python/autograph/operators/py_builtins.py:432]: Thanks for the heads up. I'll send a rollback.

Comment (lgeiger) [reply tensorflow/python/autograph/operators/py_builtins.py:432]: Good catch, sorry about that.
